integration-tests/proxy_test.go

```diff
@@ -679,17 +681,15 @@ func TestProxyNexctl(t *testing.T) {
 	require.NoError(err)
 
 	// Rules are stored
-	out, err = helper.containerExec(ctx, node1, []string{"cat", "/var/lib/nexd/proxy-rules.json"})
+	data, err := helper.containerExec(ctx, node1, []string{"cat", "/var/lib/nexd/state.json"})
 	require.NoError(err)
-	require.JSONEq(fmt.Sprintf(`
-		{
-		  "egress": [
-			"tcp:80:%s:4242"
-		  ],
-		  "ingress": [
-			"tcp:4242:127.0.0.1:8080"
-		  ]
-		}`, node1IP), out)
+	s := state.State{}
+	err = json.Unmarshal([]byte(data), &s)
+	require.NoError(err)
+	require.Equal(state.ProxyRulesConfig{
+		Egress:  []string{fmt.Sprintf("tcp:80:%s:4242", node1IP)},
```

Comment:
minor nit, `net.JoinHostPort` is best practice for joining and IP and port number. It's fine as-is here since it's an IPv4 address, but copy/paste of this code and reuse with an IPv6 address would break.